core/shared/src/main/scala/org/http4s/internal/Logger.scala

```diff
@@ -49,7 +49,7 @@ object Logger {
       val bodyStream = if (!isBinary || isJson) {
         message.bodyText(implicitly, message.charset.getOrElse(Charset.`UTF-8`))
       } else {
-        message.body.map(b => java.lang.Integer.toHexString(b & 0xff))
+        message.body.chunks.map(_.toByteVector.toHex)
```

Comment:
Nice catch! My benchmark shows that your impl has way better perf than we have now. But I accidentally tried one idea and got a bit better numbers... Wouldn't you mind if I draft PR, and we will discuss it?

Reply:
Well you can't just tease me 😛 definitely let's see it! Thanks for benching.